examples/python/images/image_reader_file.py

```diff
@@ -0,0 +1,55 @@
+###############################################################################
+# (c) 2005-2015 Copyright, Real-Time Innovations.  All rights reserved.       #
+# No duplications, whole or partial, manual or electronic, may be made        #
+# without express written permission.  Any such copies, or revisions thereof, #
+# must display this notice unaltered.                                         #
+# This code contains trade secrets of Real-Time Innovations, Inc.             #
+###############################################################################
+
+"""Reads images and saves them into a file"""
+
+from __future__ import print_function
+from sys import path as sys_path, stdout
+from os import path as os_path
+import matplotlib.pyplot as plot
+import matplotlib
+import numpy
+from time import sleep
+
+# Updating the system path is not required if you have pip-installed
+# rticonnextdds-connector
+file_path = os_path.dirname(os_path.realpath(__file__))
+sys_path.append(file_path + "/../../../")
+import rticonnextdds_connector as rti
+
+matplotlib.use('Agg') # Non-GUI backend
+
+with rti.open_connector(
+    config_name = "MyParticipantLibrary::ImageSubParticipant",
+    url = file_path + "/ImagesExample.xml") as connector:
+
+    # Create a blank image
+    fig = plot.figure()
+    blank_image = plot.imshow(numpy.zeros((40, 60, 3)))
+
+    input = connector.get_input("MySubscriber::ImageReader")
+
+    while True:
+        input.read()
+        for sample in input.valid_data_iterator:
+            # Get the received pixels (a linear sequence)
```

Comment:
Is there anyway we can combine this into a single file and just detect if there is a GUI backend or not?

Reply:
I considered that, but I prefer two scripts for two reasons:
1) The code is less complicated, and people don't necessarily need to read both scripts 
2) Someone may want to configure their system to make it work with image_reader.py, so I think I'd rather let it fail and then the user can decide whether to install the GUI backend (which depending on the Linux distro is easier or harder to do) or use image_reader_file.py